server/routes/mentions.js

```diff
@@ -0,0 +1,76 @@
+const router = require("express").Router();
+
+const requiresAuth = require("./middleware/requiresAuth");
+const { Mention, User } = require("../models");
+
+const callScraper = require("../scraper");
+
+// TODO test mentions in action
+
+router.get("/", requiresAuth, async (req, res) => {
+  const user = await User.findByPk(req.user.id);
+  // note that the getter method for n-m can accept options
+  // as if it were a regular query
+  const mentions = await user.getMentions();
+
+  res.json(mentions);
+});
+
+router.get("/:id", requiresAuth, async (req, res) => {
+  const mention = await Mention.findByPk(req.params.id);
+
+  res.json(mention);
+});
+
+router.post("/", requiresAuth, async (req, res) => {
+  // NOTE I think this is better served as just a test route in order to test the scraper
+  // This shouldn't be placed in the final PR as there will be another means to post new data
+  // into the database.
+  const user = await User.findByPk(req.user.id);
+
+  let mentions = callScraper(req.user.company, 100);
+
+  let result = [];
+  for (let m of mentions) {
+    let userMentions = await user.getMentions({
+      where: {
+        title: m.title,
+      },
+    });
+
+    // if this is new
+    if (!userMentions.length) {
+      [mention, isNew] = await Mention.findOrCreate({
```

Comment:
I think we aldready know it is new, I am wondering why `findOrCreate` instead of `create` is used here?

Reply:
I believe it is not always the case that a mention is guaranteed to be new. An example of when this might be true is when a mention references two or several different companies, and if one company were to register on our app at a later time than another company who may already be aware of this existing mention, it would be a duplicated entry on the mentions table.
That's why I opted to use `findOrCreate`; so that it links the possibly already existing entry instead of creating it twice, and thus linking it twice on the through table.

Reply:
You are right, I didn't get the idea, sorry.